src/values/AbstractObjectValue.js

```diff
@@ -498,21 +501,35 @@ export default class AbstractObjectValue extends AbstractValue {
       let generateAbstractGet = () => {
         let ob = this;
         if (this.kind === "explicit conversion to object") ob = this.args[0];
+        invariant(ob instanceof AbstractValue);
         let type = Value;
         if (P === "length" && Value.isTypeCompatibleWith(this.getType(), ArrayValue)) type = NumberValue;
-        return AbstractValue.createTemporalFromBuildFunction(
+        invariant(typeof P === "string");
+        let propertyShape, propertyGetter;
+        if (this.$Realm.instantRender.enabled && ob.shape !== undefined) {
+          propertyShape = ob.shape.getMemberAccessShapeInformation(P);
+          if (propertyShape !== undefined) {
+            propertyGetter = ob.shape.getMemberAccessGraphQLGetter(propertyShape);
+            type = propertyShape.getValueTypeForAbstract();
+          }
+        }
+        let propAbsVal = AbstractValue.createTemporalFromBuildFunction(
           this.$Realm,
           type,
           [ob],
           ([o]) => {
             invariant(typeof P === "string");
-            return memberExpressionHelper(o, P);
+            return propertyGetter !== undefined
+              ? t.callExpression(t.identifier("__prop_" + propertyGetter), [o, t.stringLiteral(P)])
```

Comment:
This will probably trip up the linter once you write some unit tests (and we need some unit tests). To address that, you want to make that `"global.__prop_"`.

Reply:
Linter shows errors with and without `global.` so I needed to define getters in my test.